### PR TITLE
egressgateway: switch to Resource[T]

### PR DESCRIPTION
**Pull request by lmb**
egressgw: use Resource[CiliumNode]

    Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

egressgw: use Resource[CiliumEndpoint]

    Replace the custom watcher with Resource[CiliumEndpoint]. This also allows
    getting rid of the custom retry queue, since Resource will by default retry
    an event if Done is called with an error.

    Signed-off-by: Lorenz Bauer <lmb@isovalent.com>

### CODE REVIEW

File: pkg/egressgateway/manager.go [hunk 1]
```diff
@@ -559,32 +498,20 @@ func (manager *Manager) deleteEndpoint(id types.NamespacedName) {
...
-	delete(manager.pendingEndpointEvents, id)
-	manager.pendingEndpointEventsLock.Unlock()
-
-	manager.endpointEventsQueue.Add(id)
+	if event.Kind == resource.Upsert {
+		event.Done(manager.addEndpoint(endpoint))
+	} else {
+		manager.deleteEndpoint(id)
+		event.Done(nil)
+	}
```
Comment (jschwinger233) [inline pkg/egressgateway/manager.go:514]: Do we need to add `manager.Lock(); defer manager.Unlock()` to protect these map insert/delete operations?

**Review by jschwinger233 — COMMENTED** [review]
Expect for one potential race, the other parts LGTM.

File: pkg/egressgateway/manager.go (hunk 1 above)
Comment (lmb) [reply pkg/egressgateway/manager.go:514]: I just had a look, manager.addEndpoint and deleteEndpoint both already take the lock.

**Review by jschwinger233 — APPROVED** [review]

**Review by joamaki — APPROVED** [review]

**Pull request merged**